### PR TITLE
Include processing log topic in cluster cleanup

### PR DESCRIPTION
**Pull request by rodesai**
### Description 

Extends the cluster cleanup handler to take a list of topics managed by the ksql
cluster that must be deleted when the cluster terminates. This gets initialized to
include the command topic, as well as the processing log topic if one is configured.

### Testing done 

added a unit test

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/api.rst [hunk 1]
```diff
@@ -330,6 +330,8 @@ If a CREATE, DROP, or TERMINATE statement returns a command status with state QU
       }
 
 
+.. _ksql_cluster_terminate:
+
 Terminate A Cluster
```
Comment (JimGalasyn) [inline docs/developer-guide/api.rst]:
```suggestion
Terminate a Cluster
```

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java [hunk 2]
```diff
@@ -33,26 +32,27 @@
...
-      final ServiceContext serviceContext
+      final ServiceContext serviceContext,
+      final List<String> managedTopics
   ) {
     Objects.requireNonNull(ksqlConfig, "ksqlConfig is null.");
     Objects.requireNonNull(ksqlEngine, "ksqlEngine is null.");
-    this.ksqlConfig = ksqlConfig;
     this.ksqlEngine = ksqlEngine;
     this.serviceContext = Objects.requireNonNull(serviceContext);
+    this.managedTopics = ImmutableList.copyOf(Objects.requireNonNull(managedTopics));
```
Comment (agavra) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java]:
```suggestion
    this.managedTopics = ImmutableList.copyOf(Objects.requireNonNull(managedTopics, "managedTopics"));
```

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java [hunk 3]
```diff
@@ -33,26 +32,27 @@
...
   }
 
   public void terminateCluster(final List<String> deleteTopicPatterns) {
     terminatePersistentQueries();
     deleteSinkTopics(deleteTopicPatterns);
-    deleteCommandTopic();
+    deleteTopics(managedTopics);
```
Comment (agavra) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java]: I'm wondering if we should close the `ksqlEngine` even if we fail to delete the topics. Thoughts?

File: docs/developer-guide/processing-log.rst [hunk 4]
```diff
@@ -158,6 +158,8 @@ The replication factor and partition count are configurable
 using the ``ksql.processing.log.topic.replication.factor`` and ``ksql.processing.log.topic.partitions`` properties,
 respectively.
 
+If ``ksql.processing.log.topic.auto.create`` is set to ``true``, the created topic will be deleted as part of :ref:`cluster termination<ksql_cluster_terminate>`.
```
Comment (agavra) [inline docs/developer-guide/processing-log.rst]: Would this mean that we lose the logs for the cluster? How would we diagnose a cluster that was accidentally terminated?

Comment (rodesai) [reply docs/developer-guide/processing-log.rst]: The processing logs, yes. Not the operational log, which is what we'd use to debug the type of issue you've described. The processing log lets users who are writing/managing ksql queries introspect what happens as ksql is processing records. So, for example, the processing log would tell a user about a deserialization failure when trying to read a row, or about the inputs/outputs to an aggregation (in debug mode). The operational log would be the place an operator would look to investigate problems running the ksql service itself (e.g. an error reading from the command topic).

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java (hunk 3 above)
Comment (rodesai) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java]: I think this should either be retry-able if it fails, or the command-runner should keep retrying on success with some polling mechanism in the API. It doesn't look like it works this way atm (it'll just leave the cluster in a partially torn down state), but we should fix that. That would be out-of-scope for this PR though.

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/RecoveryTest.java [hunk 5]
```diff
@@ -180,10 +181,9 @@ public void close() {
       this.commandRunner = new CommandRunner(
           statementExecutor,
           fakeCommandQueue,
-          ksqlConfig,
           ksqlEngine,
           1,
-          serviceContext
+          new ClusterTerminator(ksqlConfig, ksqlEngine, serviceContext, Collections.emptyList())
```
Comment (big-andy-coates) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/RecoveryTest.java]: consider just passing a mock.

**Pull request merged**